### DISCUSSION TITLE
appium-uiautomator2-driver - Mobile Gestures

### DISCUSSION THREAD

**Discussion by kiransweng** [appium/.github#1] [Q&A]
Hi Team,

I am referring latest version of appium-uiautomator2-driver , firstly i appreciate creators of this driver and document creators providing such detailed information about the driver.

I am exploring Mobile Gestures especially which is complicated functionality in the Mobile Apps Automation world.

The code examples given in this page are understandable - https://github.com/appium/appium-uiautomator2-driver/blob/master/docs/android-mobile-gestures.md but I would like to see practically to implement

I have downloaded the app - **gpsDemo-debug.apk** , and tried to install it on Emulator which is on API34. Due to compatibility issue unable to install , attached is the screenshot

<img width="252" alt="GPS-DEMO" src="https://github.com/appium/.github/assets/10164779/b680f052-613d-4427-ac80-29681b5ba6fe">

If you can point me any sutiable app that i can try and run the code snippets that would be great help.

Thanks,
Kiran Edupuganti

**Comment by mykola-mokhnach** [comment]
Try the app supplied with java client: https://github.com/appium/java-client/blob/master/src/e2eAndroidTest/resources/ApiDemos-debug.apk
